qui/updater/summary_page.py

```diff
@@ -0,0 +1,331 @@
+                             if row.vm.is_running()]
+        to_restart = [qube_row.vm
+                      for qube_row in self.list_store
+                      if qube_row.selected
+                      and qube_row.is_sys_qube]
+        to_shutdown = [qube_row.vm
+                       for qube_row in self.list_store
+                       if qube_row.selected
+                       and not qube_row.is_sys_qube]
+        shutdown_domains(tmpls_to_shutdown)
+        restart_vms(to_restart)
+        shutdown_domains(to_shutdown)
+
+
+class RestartRowWrapper(RowWrapper):
+    COLUMN_NUM = 5
+    _SELECTION = 1
+    _ICON = 2
+    _NAME = 3
+    _ADDITIONAL_INFO = 4
+
+    def __init__(self, list_store, vm, theme: Theme, _selection: Any):
+        label = QubeLabel[str(vm.label)]
+        raw_row = [
+            False,
+            load_icon(vm.icon),
+            QubeName(vm.name, label.name, theme),
+            '',
+        ]
+        super().__init__(list_store, vm, theme, raw_row)
+
+    @property
+    def selected(self):
+        return self.raw_row[self._SELECTION]
+
+    @selected.setter
+    def selected(self, value):
+        self.raw_row[self._SELECTION] = value
+        self.refresh_additional_info()
+
+    def refresh_additional_info(self):
+        self.raw_row[4] = ''
+        if self.selected and not self.is_sys_qube:
+            self.raw_row[4] = 'Restarting an app ' \
+                               'qube will shut down all running applications'
+        if self.selected and self.is_excluded:
+            self.raw_row[4] = '<span foreground="red">This qube has been ' \
+                               'explicitly disabled from restarting in ' \
+                               'settings</span>'
+
+    @property
+    def icon(self):
+        return self.raw_row[self._ICON]
+
+    @property
+    def name(self):
+        return self.vm.name
+
+    @property
+    def color_name(self):
+        return self.raw_row[self._NAME]
+
+    @property
+    def additional_info(self):
+        return self.raw_row[self._ADDITIONAL_INFO]
+
+    @property
+    def is_sys_qube(self):
+        return str(self.vm.name).startswith("sys-")
+
+    @property
+    def is_excluded(self):
+        return not get_boolean_feature(self.vm, 'restart-after-update', True)
+
+
+# TODO: duplication
+def shutdown_domains(to_shutdown):
+    """
+    Try to shut down vms and wait to finish.
+    """
+    wait_for = []
+    for vm in to_shutdown:
+        try:
+            vm.shutdown(force=True)
+            wait_for.append(vm)
+        except exc.QubesVMError:
+            pass
+
+    asyncio.run(wait_for_domain_shutdown(wait_for))
+
+    return wait_for
+
+
+# TODO: duplication
+def restart_vms(to_restart):
+    """
+    Try to restart vms.
+    """
+    shutdowns = shutdown_domains(to_restart)
+
+    # restart shutdown qubes
+    for vm in shutdowns:
+        try:
+            vm.start()
+        except exc.QubesVMError:
```

Comment:
as above: I think this should be better documented (why it's ok to ignore it) or not ignored/inform the user about errors.